lib/index.js

```diff
@@ -3055,6 +3059,254 @@ UFDS.prototype.listResolvers = function listResolvers(region, cb, noCache) {
+    } else {
+        self.getUser(user, account, _accKeySearch);
+    }
+};
+
+
+/**
+ * Loads all the access keys for a given user.
+ *
+ * @param {Object} user the user you got from getUser.
+ * @param {String} (Optional) account uuid for a customer sub-user
+ * @param {Function} callback of the form fn(err, keys).
+ * @throws {TypeError} on bad input.
+ */
+UFDS.prototype.listAccessKeys =
+function listAccessKeys(user, account, cb, noCache) {
+    if (typeof (account) === 'function') {
+        cb = account;
+        account = null;
+    }
+    if (typeof (user) !== 'string') {
+        assert.object(user, 'user');
+        if (user.account) {
+            account = user.account;
+        }
+    }
+    assert.func(cb, 'callback');
+    account = workAroundUnderspecification(account, 'account');
+
+    cb = once(cb);
+
+    var self = this;
+    function _accKeys(err, usr) {
+        if (err) {
+            cb(err);
+            return;
+        }
+        var opts = {
+            scope: 'one',
+            filter: '(objectclass=accesskey)'
+        };
+        self.search(usr.dn, opts, cb, noCache);
+    }
+
+    if (typeof (user) === 'object') {
+        _accKeys(null, user);
+    } else {
+        self.getUser(user, account, _accKeys);
+    }
+};
+
+
+/**
+ * Deletes an AccessKey under a user.
+ *
+ * @param {User} the object you got back from getUser.
+ * @param {Object} key the object you got from getAccessKey or the AccessKeyId
+ * @param {String} (Optional) account uuid for a customer sub-user
+ * @param {Function} callback of the form fn(err).
+ * @throws {TypeError} on bad input.
+ */
+UFDS.prototype.deleteAccessKey =
+function deleteAccessKey(user, key, account, cb) {
+    if (typeof (account) === 'function') {
+        cb = account;
+        account = null;
+    }
+    if (typeof (user) !== 'string') {
+        assert.object(user, 'user');
+        if (user.account) {
+            account = user.account;
+        }
+    }
+    if (typeof (key) !== 'string') {
+        assert.object(key, 'key');
+    }
+    assert.func(cb, 'callback');
+    account = workAroundUnderspecification(account, 'account');
+
+    cb = once(cb);
+
+    var self = this;
+    function _delAccKey(usr, keyObj) {
+        if (!ldap.parseDN(usr.dn).parentOf(keyObj.dn)) {
+            cb(new NotAuthorizedError(keyObj.dn + ' not a child of ' + usr.dn));
+            return;
+        }
+        self.del(keyObj.dn, cb);
+    }
+
+    function _getAccKey(usr) {
+        if (typeof (key) === 'object') {
+            _delAccKey(usr, key);
+        } else {
+            self.getAccessKey(usr, key, account, function (err, keyObj) {
+                if (err) {
+                    cb(err);
+                    return;
+                }
+                _delAccKey(usr, keyObj);
+            });
+        }
+    }
+
+    if (typeof (user) === 'object') {
+        _getAccKey(user);
+    } else {
+        this.getUser(user, account, function (err, userObj) {
```

Comment:
Maybe use the same form as previous functions, i.e.:
```
_getAccKey(null, user)
this.getUser(user, account, _getAccKey);
```

Reply:
Good call. Should probably make this change into other existing library functions.